drivers/i2c/i2c_mspm0g3xxx.c

```diff
@@ -903,7 +982,7 @@ static const struct i2c_driver_api i2c_mspm0g3xxx_driver_api = {
                                                                                                    \
 	I2C_DEVICE_DT_INST_DEFINE(index, i2c_mspm0g3xxx_init, NULL, &i2c_mspm0g3xxx_data_##index,  \
 				  &i2c_mspm0g3xxx_cfg_##index, POST_KERNEL,                        \
-				  CONFIG_I2C_INIT_PRIORITY, &i2c_mspm0g3xxx_driver_api);           \
+				  CONFIG_COUNTER_INIT_PRIORITY, &i2c_mspm0g3xxx_driver_api);           \
```

Comment:
@tbursztyka do you think this change will cause an issue?

Reply:
No you do need i2c to start after counter, as it uses the counter device at init phase. But indeed I think this change is not the "right one", I mean: there is a cleaner way to do it.

Just a sec, don't merge yet

Reply:
that's a lot of files changed here! but i definitely see your point

Reply:
> that's a lot of files changed here! but i definitely see your point.

reg-exp changing all the device instantiation macros , hopefully nothing done by hand. I should get back to it when I'll have time, only 2 tests were failing due to one external patch I was waiting for...